### PR TITLE
chore(java): add comments for loop unrolling.

### PR DESCRIPTION
**Pull request by LiangliangSui**
N/A

### CODE REVIEW

File: java/fury-core/src/main/java/org/apache/fury/memory/MemoryBuffer.java [hunk 1]
```diff
@@ -1316,24 +1316,15 @@ public int readPositiveVarInt() {
   }
 
   private int readPositiveVarIntSlow() {
-    int b = readByte();
-    int result = b & 0x7F;
-    if ((b & 0x80) != 0) {
+    int shift = 0;
```
Comment (chaokunyang) [inline java/fury-core/src/main/java/org/apache/fury/memory/MemoryBuffer.java:1319]: Could we benchmark the performance of those two implementation using JMH? The previous one are loop unrolled for performance

Comment (chaokunyang) [reply java/fury-core/src/main/java/org/apache/fury/memory/MemoryBuffer.java:1319]: The implementation in this class are kind of messy, we manually inline code and unrolled some loop for performance

Comment (chaokunyang) [reply java/fury-core/src/main/java/org/apache/fury/memory/MemoryBuffer.java:1319]: Yes, could you add some comments in this pr? I think we'd better add comments to similar methods

Comment (LiangliangSui) [reply java/fury-core/src/main/java/org/apache/fury/memory/MemoryBuffer.java]: Updated.

BTW, do we need to add the benchmark code (sample code above) that uses loop and does not use loop in the `MemoryBuffer#readPositiveVarIntSlow` method to the benchmark directory?

Comment (chaokunyang) [reply java/fury-core/src/main/java/org/apache/fury/memory/MemoryBuffer.java]: I don't think it's necessary, only rolling code will be used in the release

**Comment by chaokunyang** [comment]
@LiangliangSui There are some conflict between your branch and main branch, could you resolve the conflict?

**Comment by LiangliangSui** [comment]
> @LiangliangSui There are some conflict between your branch and main branch, could you resolve the conflict?

Updated, ptal.

**Review by chaokunyang — APPROVED** [review]
LGTM

**Pull request merged**